kustomize edit add label produces panics

Adding labels like `kustomize edit add label foo:bar` panics in v1.0.8. Am I doing something stupid?

Reproducer for both linux and mac:

```sh
sh <<'SCRIPT'
set -e
export CDPATH=''

WORKDIR="$(mktemp -d)"
trap "rm -rf -- '$WORKDIR'" INT EXIT

cd -- "$WORKDIR"

curl -fLo "kustomize" "https://github.com/kubernetes-sigs/kustomize/releases/download/v1.0.8/kustomize_1.0.8_$(uname | tr '[:upper:]' '[:lower:]')_amd64"
chmod +x "kustomize"

cat <<EOF >"deployment.yaml"
apiVersion: apps/v1
kind: Deployment
metadata:
  name: pause
spec:
  template:
    spec:
      containers:
      - name: pause
        image: gcr.io/google_containers/pause-amd64:3.1
EOF

touch kustomization.yaml
./kustomize edit add resource deployment.yaml
./kustomize edit add label foo:bar
./kustomize build
SCRIPT
```

The panic looks like this:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x96a608]

goroutine 1 [running]:
sigs.k8s.io/kustomize/pkg/validators.MakeLabelValidator.func1(0xc420195500, 0x7ffd1e5bbc85, 0x7)
        /go/src/sigs.k8s.io/kustomize/pkg/validators/validators.go:30 +0x98
sigs.k8s.io/kustomize/pkg/commands.(*addMetadataOptions).validateAndParse(0xc420502980, 0xc42037fad0, 0x1, 0x1, 0xc42033dcf0, 0x5825e0)
        /go/src/sigs.k8s.io/kustomize/pkg/commands/addmetadata.go:122 +0x88
sigs.k8s.io/kustomize/pkg/commands.(*addMetadataOptions).runE(0xc420502980, 0xc42037fad0, 0x1, 0x1, 0x135d640, 0x1a9cde8, 0xc42033dce0, 0xc42033dcd8, 0xc
42033dca8)
        /go/src/sigs.k8s.io/kustomize/pkg/commands/addmetadata.go:91 +0x4d
sigs.k8s.io/kustomize/pkg/commands.newCmdAddLabel.func1(0xc4201e0c80, 0xc42037fad0, 0x1, 0x1, 0x0, 0x0)
        /go/src/sigs.k8s.io/kustomize/pkg/commands/addmetadata.go:83 +0x7c
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).execute(0xc4201e0c80, 0xc42037fa80, 0x1, 0x1, 0xc4201e0c80, 0xc42037fa80)
        /go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:756 +0x468
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42010e280, 0xc42010ea00, 0xc42010ec80, 0xc420278280)
        /go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:846 +0x30a
sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42010e280, 0x1289d78, 0xc420094058)
        /go/src/sigs.k8s.io/kustomize/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.main()
        /go/src/sigs.k8s.io/kustomize/kustomize.go:30 +0x43
```

--- a/pkg/validators/validators.go
+++ b/pkg/validators/validators.go
@@ -15,7 +15,7 @@ type MapValidatorFunc func(map[string]string) error
 func MakeAnnotationValidator() MapValidatorFunc {
 	return func(x map[string]string) error {
 		errs := apivalidation.ValidateAnnotations(x, field.NewPath("field"))
-		if errs != nil {
+		if len(errs) > 0 {
 			return errors.New(errs.ToAggregate().Error())
 		}
 		return nil
@@ -26,7 +26,7 @@ func MakeAnnotationValidator() MapValidatorFunc {
 func MakeLabelValidator() MapValidatorFunc {
 	return func(x map[string]string) error {
 		errs := v1validation.ValidateLabels(x, field.NewPath("field"))
-		if errs != nil {
+		if len(errs) > 0 {
 			return errors.New(errs.ToAggregate().Error())
 		}
 		return nil

